codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -276,10 +277,17 @@ class HttpBindingGenerator(
                         }
                     }
                     if (targetShape.hasTrait<EnumTrait>()) {
-                        rust(
-                            "Ok(#T::from(body_str))",
-                            symbolProvider.toSymbol(targetShape)
-                        )
+                        if (mode == CodegenMode.Server) {
```

Comment:
I am not a fan of all these if `CodegenMode.Server` / `CodegenMode.Client` :(

Reply:
It's because client and server only change for one line or two every time; otherwise we'd have to create a new class every time (or two) and copy the remaining code. But if that's the best approach for this, I'll go for it